api/v1beta1/kogitobuild_types.go

```diff
@@ -181,58 +324,51 @@ type KogitoBuildStatus struct {
 	Builds Builds `json:"builds"`
 }
 
-// KogitoBuildConditionType ...
-type KogitoBuildConditionType string
+// GetLatestBuild ...
+func (k *KogitoBuildStatus) GetLatestBuild() string {
+	return k.LatestBuild
+}
 
-const (
-	// KogitoBuildSuccessful condition for a successful build.
-	KogitoBuildSuccessful KogitoBuildConditionType = "Successful"
-	// KogitoBuildFailure condition for a failure build.
-	KogitoBuildFailure KogitoBuildConditionType = "Failed"
-	// KogitoBuildRunning condition for a running build.
-	KogitoBuildRunning KogitoBuildConditionType = "Running"
-)
+// SetLatestBuild ...
+func (k *KogitoBuildStatus) SetLatestBuild(latestBuild string) {
+	k.LatestBuild = latestBuild
+}
 
-// KogitoBuildConditionReason ...
-type KogitoBuildConditionReason string
+// GetConditions ...
+func (k *KogitoBuildStatus) GetConditions() []api.KogitoBuildConditionsInterface {
+	var conditions []api.KogitoBuildConditionsInterface
+	for _, condition := range k.Conditions {
+		conditions = append(conditions, &condition)
```

Comment:
Appending pointer of a value from range is actually bad, as value reference is not changed between iterations. The result is that all elements of the conditions slice point to same value.
Try for example this test:
```
func TestKogitoBuildStatus(t *testing.T) {
	instance := &v1beta1.KogitoBuild{
		Status: v1beta1.KogitoBuildStatus{
			Conditions: []v1beta1.KogitoBuildConditions{
				{
					Type: api.KogitoBuildSuccessful,
				},
				{
					Type: api.KogitoBuildFailure,
				},
			},
		},
	}

	cond := instance.GetStatus().GetConditions()
	assert.NotEqual(t, cond[0], cond[1])
}
```
It needs to be fixed for all occurrences.

Reply:
done